week1/search.py

```diff
@@ -99,13 +107,82 @@ def query():
 
 def create_query(user_query, filters, sort="_score", sortDir="desc"):
     print("Query: {} Filters: {} Sort: {}".format(user_query, filters, sort))
-    query_obj = {
-        "size": 10,
-        "query": {
-            "match_all": {}  # Replace me with a query that both searches and filters
-        },
-        "aggs": {
-            # TODO: FILL ME IN
+    query_obj = {"size": 10, "track_total_hits": True, "query": {}}
+
+    match_clause = None
+    if user_query and user_query != "*":
+        match_clause = {
+            "bool": {
+                "must": [
+                    {
+                        "multi_match": {
+                            "query": user_query,
+                            "fields": [
+                                "name^100",
+                                "shortDescription^50",
+                                "longDescription^10",
+                                "department",
+                            ],
+                        }
+                    }
+                ],
+                "filter": filters,
+            }
+        }
+    elif filters:
+        match_clause = {
+            "bool": {
+                "filter": filters,
+            }
+        }
+    else:
+        match_clause = {"match_all": {}}
```

Comment:
instead of handling if-else, let ElasticSearch handle this.
We can achieve this by having bool should adding multi_match and match_all query and saying "minimum_should_match" = 1